src/Build/Evaluation/Expander.cs

```diff
@@ -2313,6 +2313,10 @@ internal static ItemTransformFunction GetItemTransformFunction(IElementLocation
                                 rootedPath = Path.Combine(baseDirectoryToUse, unescapedPath);
                             }
 
+                            // Run the path through GetFullPath to remove path elements like "..".
+                            // Otherwise we run the risk of returning two or more different paths that
+                            // represent the same directory.
+                            rootedPath = Path.GetFullPath(rootedPath);
```

Comment:
Is `Path.GetFullPath` the best option here, or should I be using something like `FileUtilities.NormalizePath`?

Reply:
Ah, good point. Use `FileUtilities.NormalizePath`, which jumps through some hoops on Windows/full framework to support long paths if possible.